themes/classic/templates/catalog/_partials/product-cover-thumbnails.tpl

```diff
@@ -36,18 +36,19 @@
             alt="{$product.name}"
           {/if}
           loading="lazy"
-          width="452"
-          height="452"
+          width="{$product.default_image.bySize.medium_default.width}"
+          height="{$product.default_image.bySize.medium_default.height}"
         >
         <div class="layer hidden-sm-down" data-toggle="modal" data-target="#product-modal">
           <i class="material-icons zoom-in">search</i>
         </div>
       {else}
-        <img 
+        <img
+          class="img-fluid"
           src="{$urls.no_picture_image.bySize.large_default.url}"
           loading="lazy"
-          width="452"
-          height="452"
+          width="{$product.default_image.bySize.medium_default.width}"
```

Comment:
```suggestion
          width="{$product.default_image.bySize.large_default.width}"
```
Same question here

Reply:
it would give a width and height of 800.... the image sizes don't match the size we get from the product object